rackcity/views/asset_views.py

```diff
@@ -239,58 +239,61 @@ def asset_add(request):
             {"failure_message": Status.CREATE_ERROR.value + str(error)},
             status=HTTPStatus.BAD_REQUEST,
         )
-    if serializer.validated_data["model"].is_rackmount():
-        if (
-            "rack" not in serializer.validated_data
-            or not serializer.validated_data["rack"]
-            or "rack_position" not in serializer.validated_data
-            or not serializer.validated_data["rack_position"]
-        ):
-            return JsonResponse(
-                {
-                    "failure_message": Status.INVALID_INPUT.value
-                    + "Must include rack and rack position to add a rackmount asset. "
-                },
-                status=HTTPStatus.BAD_REQUEST,
-            )
-        rack_id = serializer.validated_data["rack"].id
-        rack_position = serializer.validated_data["rack_position"]
-        height = serializer.validated_data["model"].height
-        try:
-            validate_asset_location_in_rack(
-                rack_id, rack_position, height, change_plan=change_plan
-            )
-        except LocationException as error:
-
-            return JsonResponse(
-                {"failure_message": Status.CREATE_ERROR.value + str(error)},
-                status=HTTPStatus.BAD_REQUEST,
-            )
-    else:
-        if (
-            "chassis" not in serializer.validated_data
-            or not serializer.validated_data["chassis"]
-            or "chassis_slot" not in serializer.validated_data
-            or not serializer.validated_data["chassis_slot"]
-        ):
-            return JsonResponse(
-                {
-                    "failure_message": Status.INVALID_INPUT.value
-                    + "Must include chassis and chassis slot to add a blade asset. "
-                },
-                status=HTTPStatus.BAD_REQUEST,
-            )
-        chassis_id = serializer.validated_data["chassis"].id
-        chassis_slot = serializer.validated_data["chassis_slot"]
-        try:
-            validate_asset_location_in_chassis(
-                chassis_id, chassis_slot, change_plan=change_plan
-            )
-        except LocationException as error:
-            return JsonResponse(
-                {"failure_message": Status.CREATE_ERROR.value + str(error)},
-                status=HTTPStatus.BAD_REQUEST,
-            )
+    if not (
+        "offline_storage_site" in serializer.validated_data
+        and serializer.validated_data["offline_storage_site"]
+    ):
```

Comment:
asset only needs to have (rack+rack position) or (chassis+chassis slot) if it's NOT in storage